internal/incident/incident.go

```diff
@@ -429,24 +427,15 @@ func (i *Incident) evaluateEscalations(eventTime time.Time) ([]*rule.Escalation,
 		// Check if new escalation stages are reached
 		for _, escalation := range r.Escalations {
 			if _, ok := i.EscalationState[escalation.ID]; !ok {
-				matched := false
-
-				if escalation.Condition == nil {
-					matched = true
-				} else {
-					var err error
-					matched, err = escalation.Condition.Eval(filterContext)
-					if err != nil {
-						i.logger.Warnw(
-							"Failed to evaluate escalation condition", zap.String("rule", r.Name),
-							zap.Object("escalation", escalation), zap.Error(err),
-						)
-
-						matched = false
-					} else if !matched {
-						incidentAgeFilter := filterContext.ReevaluateAfter(escalation.Condition)
-						retryAfter = min(retryAfter, incidentAgeFilter)
-					}
+				matched, err := escalation.Eval(filterContext)
+				if err != nil {
+					i.logger.Warnw(
+						"Failed to evaluate escalation condition", zap.Object("rule", r),
+						zap.Object("escalation", escalation), zap.Error(err),
+					)
+				} else if !matched {
+					incidentAgeFilter := filterContext.ReevaluateAfter(escalation.Condition)
+					retryAfter = min(retryAfter, incidentAgeFilter)
 				}
 
 				if matched {
```

Comment:
The existing `if` below can be combined with the if before. That's also a bit nicer because then `matched` will simply be ignored if an error was returned.